runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRLogEntry.java

```diff
@@ -0,0 +1,129 @@
+package org.corfudb.protocols.logprotocol;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ * A log entry structure which contains a collection of multiSMRentries,
+ * each one contains a list of updates for one object.
+ */
+@SuppressWarnings("checkstyle:abbreviation")
+@ToString
+@Slf4j
+public class SMRLogEntry extends LogEntry {
+
+    // Map from stream-ID to a list of SMR updates to this stream.
+    @Getter
+    public Map<UUID, List<SMRRecord>> entryMap = Collections.synchronizedMap(new HashMap<>());
+
+    public SMRLogEntry() {
+        this.type = LogEntryType.SMRLOG;
+    }
+
+    public SMRLogEntry(Map<UUID, List<SMRRecord>> entryMap) {
+        this.type = LogEntryType.SMRLOG;
+        this.entryMap = entryMap;
+    }
+
+    /**
+     * Extract a particular stream's entry from this object.
+     *
+     * @param streamID stream ID
+     * @return the SMR Record list corresponding to streamId
+     */
+    private List<SMRRecord> getStreamEntry(UUID streamID) {
```

Comment:
check null

Reply:
This is private API, I think it's not that necessary since the public APIs are not even doing this.